src/util.ts

```diff
@@ -953,3 +989,27 @@ export function isTokenDetectionEnabledForNetwork(chainId: string): boolean {
     chainId,
   );
 }
+
+/**
+ * Utility method to log if error is a common fetch error and otherwise rethrow it.
+ *
+ * @param error - Caught error that we should either rethrow or log to console.
+ * @param codesToCatch - array of error codes for errors we want to catch and log in a particular context
+ */
+export function logOrRethrowError(error: any, codesToCatch?: number[]) {
+  const includesErrorCodeToCatch = codesToCatch?.length
+    ? codesToCatch?.some((code) =>
+        error.message?.includes(`Fetch failed with status '${code}'`),
+      )
+    : false;
```

Comment:
`some` already returns a boolean (and returns false if the array is empty). Also, instead of having to check for `codesToCatch`, what do you think about defaulting it to an empty array? So this would be:

``` suggestion
export function logOrRethrowError(error: any, codesToCatch?: number[] = []) {
  const includesErrorCodeToCatch = codesToCatch.some((code) =>
    error.message?.includes(`Fetch failed with status '${code}'`),
  );
```

Reply:
done here: https://github.com/MetaMask/controllers/pull/805/commits/180b2802d9edef8ff9392b88ea3a3f76cc3373da